### PR TITLE
added Fedora installation items

### PR DESCRIPTION
**Pull request by Tigro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,15 @@ $ yay -S python-validity
 $ fprintd-enroll
 ```
 
+On Fedora Linux
+
+```
+$ sudo dnf copr enable tigro/python-validity
+$ sudo apt install open-fprintd fprintd-clients fprintd-clients-pam python3-validity
```
Comment (cewbdex) [inline README.md:28]: apt on Fedora?

**Review by cewbdex — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Tigro) [reply README.md]: You're right. Fix to dnf.

**Comment by uunicorn** [comment]
Thanks!

**Comment by rmeissn** [comment]
I've tried to install it this way on the latest Fedora version and dnf tells me about that no job control is available. The package failed to install correctly (but is tracked as installed). I can't remove the package either, as it fails because of the same reason: "no job control in this shell". How can I install/remove the package correctly?

**Comment by Tigro** [comment]
> I've tried to install it this way on the latest Fedora version and dnf tells me about that no job control is available. The package failed to install correctly (but is tracked as installed). I can't remove the package either, as it fails because of the same reason: "no job control in this shell". How can I install/remove the package correctly?

Hello Roy, can you attach more information, such as commands (bash/dnf) and full output. In this case, it will be easier for me to  investigate you problem.

**Comment by rmeissn** [comment]
@Tigro Yeah, sure. As my system is set to german language I translated the given information below with google.
I precisely followed the described steps in the readme file.
```
>>>sudo dnf remove python3-validity                                                  
Dependencies are resolved.
============================================================================= 
 Package Architecture Version Repository Size
============================================================================= 
Remove:
 python3-validity noarch 0.12-1.fc34 @@ System 697 k
Dependencies that are not required are removed:
 innoextract x86_64 1.9-3.fc34 @fedora 582 k
 python3-pyusb noarch 1.0.2-11.fc34 @fedora 375 k

Transaction overview
=================================================== 
Remove 3 packages

Free space: 1.6 M.
Is this ok? [y / N]: y
Transaction verification is in progress
Error: Transaction Validation Conflicts Dependency Resolution:
innoextract is required by (installed) python3-validity-0.12-1.fc34.noarch
python3.9dist (pyusb)> = 1 is required by (installed) python3-validity-0.12-1.fc34.noarch
To investigate the problem, try running: "rpm -Va --nofiles --nodigest".
Your RPM database may be corrupted, "rpm --rebuilddb" could fix the problem.
```
Judging from running several commands my RPM database seems fine.

```
>>>sudo dnf remove python3-validity --exclude = "innoextract" --exclude = "python3-pyusb"
Dependencies are resolved.
=================================================== 
 Package Architecture Version Repository Size
=================================================== 
Remove:
 python3-validity noarch 0.12-1.fc34 @@ System 697 k

Transaction overview
=================================================== 
Remove 1 package

Free space: 697 k
Is this ok? [y / N]: y
Transaction verification is in progress
Transaction check was successful.
Transaction is being tested
Transaction test was successful.
Transaction is in progress
  Preparation in progress: 1/1
  Executed scriptlet: python3-validity-0.12-1.fc34.noarch 1/1
/var/tmp/rpm-tmp.c2nMcq: Line 1: fg: No job control in this shell.
Error:% preun (python3-validity-0.12-1.fc34.noarch) scriptlet failed, exit status 1

Error in PREUN scriptlet in rpm package python3-validity
  Checking: python3-validity-0.12-1.fc34.noarch 1/1

Failed:
  python3-validity-0.12-1.fc34.noarch

Error: transaction failed
```

That's basically the same error as the installation of the package gave me:
```
>>>sudo dnf reinstall python3-validity
Last check for expired metadata: before 1:31:40 on Tue 15 Jun 2021 14:11:21 CEST.
Dependencies are resolved.
=================================================== 
 Package Architecture Version Repository Size
=================================================== 
Reinstall:
 python3-validity noarch 0.12-1.fc34 copr: copr.fedorainfracloud.org: tigro: python-validity 136 k
     Replace python3-validity.noarch 0.12-1.fc34

Transaction overview
=================================================== 

Total download size: 136 k
Is this ok? [y / N]: y
Packages are downloaded:
python3-validity-0.12-1.fc34.noarch.rpm 249 kB / s | 136 kB 00:00
-------------------------------------------------- 
Total 247 kB / s | 136 kB 00:00
Transaction verification is in progress
Transaction check was successful.
Transaction is being tested
Transaction test was successful.
Transaction is in progress
  Preparation in progress: 1/1
  Reinstall: python3-validity-0.12-1.fc34.noarch 1/3
  Executed scriptlet: python3-validity-0.12-1.fc34.noarch 1/3
/var/tmp/rpm-tmp.4KW6YF: Line 1: fg: No job control in this shell.
Traceback (most recent call last):
  File "/ bin / validity-sensors-firmware", line 92, in <module>
    raise exception ('No supported validity device found')
Exception: No supported validity device found
/var/tmp/rpm-tmp.4KW6YF: Line 6: fg: No job control in this shell.
Warning:% post (python3-validity-0.12-1.fc34.noarch) scriptlet failed, exit status 1

Error in POSTIN scriptlet in rpm package python3-validity
  Executed scriptlet: python3-validity-0.12-1.fc34.noarch 2/3
/var/tmp/rpm-tmp.bvkbHi: Line 1: fg: No job control in this shell.
Error:% preun (python3-validity-0.12-1.fc34.noarch) scriptlet failed, exit status 1

Error in PREUN scriptlet in rpm package python3-validity
  Executed scriptlet: python3-validity-0.12-1.fc34.noarch 3/3
Error: python3-validity-0.12-1.fc34.noarch: delete failed
/var/tmp/rpm-tmp.IUWjBG: Line 1: fg: No job control in this shell.
Error:% preun (python3-validity-0.12-1.fc34.noarch) scriptlet failed, exit status 1

Error in PREUN scriptlet in rpm package python3-validity
Error: python3-validity-0.12-1.fc34.noarch: delete failed

  Validation in progress: python3-validity-0.12-1.fc34.noarch 1/3
  Review lRuns: python3-validity-0.12-1.fc34.noarch 2/3
  Checking in progress: python3-validity-0.12-1.fc34.noarch 3/3

Reinstalled:
  python3-validity-0.12-1.fc34.noarch
Failed:
  python3-validity-0.12-1.fc34.noarch python3-validity-0.12-1.fc34.noarch

Error: transaction failed
```

This is the same error which occurred when installing all the packages, listed within the readme file.

**Comment by Tigro** [comment]
Ok. Some errors in %preun scriptlets but "/var/tmp/rpm-tmp.O6Mnk0: line 1: fg: no job control" not a problem.

You can check rpm -q python3-validity. I think many versions are in the rpm base. To fix it try to remove

# rpm -e python3-validity --allmatches --noscripts
# rpm -e open-fprintd --allmatches --noscripts

open-fprintd also have this bug

and after this install from repo **NEW** versions of python3-validity and open-fprintd.

**Comment by rmeissn** [comment]
Thanks for the command suggestions. I managed to remove and later install the mentioned packages. Installation still complained about missing job control.
Enrolling and verifying fingerprints works on the CLI, but the fingerprint reader is not recognized by gnome (or any other authentication method) - service is started. As `pam-auth-update` is not available on Fedora 34, I'm not sure how to proceed... Do you have an advice for me?

**Comment by Tigro** [comment]
I applied these commands yesterday to enable fingerprint in GNOME

```
$ sudo authselect enable-feature with-fingerprint
$ sudo authselect apply-changes
```

after that I reboot laptop

**Comment by alexjfinch** [comment]
Thanks for this - I had to reinstall Fedora 34 from scratch and the copr repo was certainly easier and cleaner than the installation I had before, however still running into issues getting this working.

Now systemd won't start the process and gives an error that the start request is repeated too quickly? I tried to edit the service with different restart comments, but failed to get it to run. Any ideas? (I'm a total noob btw)....

At least with the most recent update the other permission issues have now been resolved! thanks!

$ sudo systemctl status python3-validity 
× python3-validity.service - python-validity driver dbus service
     Loaded: loaded (/usr/lib/systemd/system/python3-validity.service; enabled; vendor preset: disabled)
     Active: failed (Result: exit-code) since Thu 2021-06-24 13:15:38 BST; 1min 33s ago
    Process: 3991 ExecStart=/usr/lib/python-validity/dbus-service --debug (code=exited, status=1/FAILURE)
   Main PID: 3991 (code=exited, status=1/FAILURE)
        CPU: 136ms

Jun 24 13:15:38 systemd[1]: python3-validity.service: Scheduled restart job, restart counter is at 5.
Jun 24 13:15:38 systemd[1]: Stopped python-validity driver dbus service.
Jun 24 13:15:38 systemd[1]: python3-validity.service: Start request repeated too quickly.
Jun 24 13:15:38 systemd[1]: python3-validity.service: Failed with result 'exit-code'.
Jun 24 13:15:38 systemd[1]: Failed to start python-validity driver dbus service.

**Comment by Tigro** [comment]
Hi!

Try to see log by

journalctl -u python3-validity.service

**Comment by alexjfinch** [comment]
> Hi!
> 
> Try to see log by
> 
> journalctl -u python3-validity.service
> 
> and try Error section from
> https://github.com/uunicorn/python-validity#error-situations

Thanks - I tried the error section to no avail.

Looking at journalctl its actually throwing up the permission error for the backoff file again - can you remember how that one was solved?

**Comment by Tigro** [comment]
Some instruction are here https://github.com/uunicorn/python-validity/issues/54

Maybe setenforce 0 can help your.

**Comment by alexjfinch** [comment]
sudo chmod 777 /usr/share/python3-validity seemed to work, although I don't like the thought of 777 on that folder.

Is there a better user permission level to use that providing all access?

**Comment by Tigro** [comment]
Check selinux restrictions first. Install setroubleshoot package and run

```
sealert
```

**Comment by alexjfinch** [comment]
Ok fixed, its the SELinux policy that needs to be compiled and added.

The .te file is already on the git page and you just need to convert it into .mod and then .pp and enable it (or whatever the term is).

Can these instructions be added to the fedora instructions as I have no clue how to update! I guess also the bit about authselect I wrote out in a previous post to enable fingerprint

**Comment by dbasinge** [comment]
> Ok. Some errors in %preun scriptlets but "/var/tmp/rpm-tmp.O6Mnk0: line 1: fg: no job control" not a problem.
> 
> You can check rpm -q python3-validity. I think many versions are in the rpm base. To fix it try to remove
> 
> ```
> # rpm -e python3-validity --allmatches --noscripts
> # rpm -e open-fprintd --allmatches --noscripts
> ```
> 
> open-fprintd also have this bug
> 
> and after this install from repo **NEW** versions of python3-validity and open-fprintd.

Still getting this with open-fprintd-0.6-2.fc34.noarch and python3-validity-0.12-4.fc34.noarch

**Comment by Tigro** [comment]
Ohhhh. I understood this problem. After build in copr, macros in builded rpm-package not expanded. They like:

```
postinstall scriptlet (using /bin/sh):
%systemd_post open-fprintd.service || true
preuninstall scriptlet (using /bin/sh):
%systemd_preun open-fprintd.service || true
postuninstall scriptlet (using /bin/sh):
%systemd_postun_with_restart open-fprintd.service
```

If I build package on localhost, macros are:

```
postinstall scriptlet (using /bin/sh):


if [ $1 -eq 1 ] && [ -x /usr/bin/systemctl ]; then
    # Initial installation
    /usr/bin/systemctl --no-reload preset open-fprintd.service || true || :
fi
preuninstall scriptlet (using /bin/sh):


if [ $1 -eq 0 ] && [ -x /usr/bin/systemctl ]; then
    # Package removal, not upgrade
    if [ -d /run/systemd/system ]; then
          /usr/bin/systemctl --no-reload disable --now open-fprintd.service || true || :
    else
          /usr/bin/systemctl --no-reload disable open-fprintd.service || true || :
    fi
fi
postuninstall scriptlet (using /bin/sh):


if [ $1 -ge 1 ] && [ -x /usr/bin/systemctl ]; then
    # Package upgrade, not uninstall
    for unit in open-fprintd.service; do
         /usr/bin/systemctl set-property $unit Markers=+needs-restart || :
    done
fi
```

I will try to find a solution.

**Comment by Tigro** [comment]
Yes. Resolved in 

```
python3-validity-0.12-5
open-fprintd-0.6-5.
```

by adding
```
BuildRequires:  systemd-rpm-macros
```
in spec file.

But as installed packages are broken you should remove it and install new versions.

**Pull request merged**